src/snowflake/cli/_plugins/streamlit/streamlit_entity_model.py

```diff
@@ -43,24 +44,22 @@ class StreamlitEntityModel(EntityModelBase, ExternalAccessBaseModel, ImportsBase
     stage: Optional[str] = Field(
         title="Stage in which the app’s artifacts will be stored", default="streamlit"
     )
-    # Possibly can be PathMapping
-    artifacts: Optional[List[Path]] = Field(
+    artifacts: Optional[List[Union[PathMapping, Path, str]]] = Field(
         title="List of files which should be deployed. Each file needs to exist locally. "
         "Main file needs to be included in the artifacts.",
         default=None,
     )
 
-    @model_validator(mode="after")
-    def artifacts_must_exists(self):
-        if not self.artifacts:
-            return self
-
-        for artifact in self.artifacts:
-            if "*" in artifact.name:
-                continue
-            if not artifact.exists():
-                raise ValueError(
-                    f"Specified artifact {artifact} does not exist locally."
-                )
-
-        return self
+    @field_validator("artifacts")
```

Comment:
Is this something that will stay, or can be removed in 4.0?

Reply:
It will stay, same thing is done in Snowpark. Validator changes string to PathMapping objects.